File: 121.best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:

        min_price, max_profit = float('inf'), 0
        for price in prices:
            min_price = min(min_price, price)
            profit = price - min_price
            max_profit = max(profit, max_profit)


        if max_profit <= 0:
            return 0
        else:
            return max_profit

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int max_profit = 0;
        int curr_min = 1 << 30;
        for (int i=0;i<prices.size();i++) {
            max_profit = max(prices[i] - curr_min, max_profit);
            if (prices[i] < curr_min) {
                curr_min = prices[i];
            }
        }
        return max_profit;
        
    }
};